Version of the patch of the same name from sato-icon-theme JL - 20/10/10
"
pkg-config will only search the target sysroot and we want the native script. This 
patch adds an option to allow the path to the tool to be specified.

RP - 12/8/10"

Upstream-Status: Pending

Index: gnome-icon-theme-2.31.0/configure.ac
===================================================================
--- gnome-icon-theme-2.31.0.orig/configure.ac	2010-12-01 11:00:22.000000000 +0800
+++ gnome-icon-theme-2.31.0/configure.ac	2010-12-01 11:02:16.000000000 +0800
@@ -38,18 +38,24 @@
 ICONMAP="true"
 if test "x$enable_mapping" != "xno"; then
    UTILS_REQUIRED=0.8.7
+   PKG_PROG_PKG_CONFIG()
 
-   AC_MSG_CHECKING([icon-naming-utils >= $UTILS_REQUIRED])
-   PKG_CHECK_EXISTS(icon-naming-utils >= $UTILS_REQUIRED,
-                    have_utils=yes, have_utils=no)
-   if test "x$have_utils" = "xyes"; then
-      UTILS_PATH="`$PKG_CONFIG --variable=program_path icon-naming-utils`"
-      ICONMAP="$UTILS_PATH/icon-name-mapping"
-      AC_MSG_RESULT([yes])
-   else
-      AC_MSG_RESULT([no])
-      AC_MSG_ERROR([icon-naming-utils >= $UTILS_REQUIRED is required to build
-                    and install gnome-icon-theme])
+   AC_ARG_WITH(iconmap,
+   	    AC_HELP_STRING([--with-iconmap=<dir>], [The location of the icon-name-mapping script to use]),
+   	    ICONMAP=$withval, ICONMAP="")
+   if test "x$ICONMAP" = "x"; then
+      AC_MSG_CHECKING([icon-naming-utils >= $UTILS_REQUIRED])
+      PKG_CHECK_EXISTS(icon-naming-utils >= $UTILS_REQUIRED,
+      		    have_utils=yes, have_utils=no)
+      if test "x$have_utils" = "xyes"; then
+         UTILS_PATH="`$PKG_CONFIG --variable=program_path icon-naming-utils`"
+         ICONMAP="$UTILS_PATH/icon-name-mapping"
+
+         AC_MSG_RESULT([yes])
+      else
+         AC_MSG_RESULT([no])
+         AC_MSG_ERROR([icon-naming-utils >= $UTILS_REQUIRED is required to build and install sato-icon-theme"])
+      fi
    fi
 else
    ICONMAP="false"
